src/main/java/cms/gov/madie/measure/resources/MeasureTransferController.java

```diff
@@ -101,9 +112,22 @@ public ResponseEntity<Measure> createMeasure(
     // set ids for groups
     measure.getGroups().forEach(group -> group.setId(ObjectId.get().toString()));
     reorderGroupPopulations(measure.getGroups());
-    Measure savedMeasure = repository.save(measure);
-    measureSetService.createMeasureSet(
-        harpId, savedMeasure.getId(), savedMeasure.getMeasureSetId());
+
+    Measure savedMeasure = null;
+    if (!CollectionUtils.isEmpty(measuresWithSameSetId)) {
```

Comment:
As a result of my earlier suggestion to move the outer if, this now needs a model check, too. Otherwise, we'll delete and overwrite QI-Core and QDM measures.